aicsimageio/writers/bfio_writers.py

```diff
@@ -0,0 +1,153 @@
+            of the form "Channel:channel_index"
+        physical_pixel_sizes: Optional[Union[types.PhysicalPixelSizes,
+                List[types.PhysicalPixelSizes]]]
+            List of numbers representing the physical pixel sizes in Z, Y, X in microns
+            Default: None
+
+        Note
+        ----
+        This writer can perform chunked writing. When the input array is a dask array or
+        an xarray DataArray backed by dask, data is written one XY plane at a time to
+        conserve memory. This is especially useful when the image will not fit into
+        memory. The underlying write can support further chunking in XY dimensions, so
+        this functionality should be added in the future.
+
+        Raises
+        ------
+        ValueError:
+            Non-local file system URI provided.
+
+        Examples
+        --------
+        Write a TCZYX data set to a tiled OME-Tiff
+
+        >>> image = numpy.ndarray([1, 10, 3, 1024, 2048])
+        ... OmeTiledTiffWriter.save(image, "file.ome.tif")
+
+        If an image is too large to fit into memory, a dask array can be passed in and
+        chunked writing will take place. This will work if each YX plane can fit into
+        memory.
+
+        >>> image = numpy.ndarray([1, 10, 3, 1024, 2048])
+        ... OmeTiledTiffWriter.save(image, "file.ome.tif")
+        """
+        if data.ndim > 5 or data.ndim < 2:
+            raise ValueError(
+                "Data must have 2-5 dimensions and be in TCZYX order. "
+                + "The S (sample) dimension is not allowed."
+            )
+
+        # Resolve final destination
+        fs, path = io_utils.pathlike_to_fs(uri)
+
+        # Catch non-local file system
+        if not isinstance(fs, LocalFileSystem):
+            raise ValueError(
+                f"Cannot write to non-local file system. "
+                f"Received URI: {uri}, which points to {type(fs)}."
+            )
+
+        # If metadata is attached as lists, enforce matching shape
+        if not isinstance(data, (np.ndarray, da.core.Array)):
+            raise TypeError(
+                "Input data for the OmeTiledTiffWriter must be a numpy.ndarray"
+            )
+
+        if isinstance(ome_xml, str):
+            ome_xml = from_xml(ome_xml)
+
+        with BioWriter(path, metadata=ome_xml, backend="python") as bw:
+
+            # Define xml if ome_xml is not present but other kwargs are
+            if ome_xml is None:
+                if channel_names is not None:
+                    bw.channel_names = channel_names
+
+                if physical_pixel_sizes is not None:
+                    bw.ps_z = (physical_pixel_sizes[0], None)
+                    bw.ps_y = (physical_pixel_sizes[1], None)
+                    bw.ps_x = (physical_pixel_sizes[2], None)
+
+                for dim, val in zip("XYZCT", reversed(data.shape)):
+                    setattr(bw, dim, val)
```

Comment:
Could we use / should we use the OmeTiffWriter build_ome function here instead?

Reply:
I can look into it, but without looking into it I would say no. The reason being that we intertwine the OME metadata with the class attributes, which we also use when writing out the data to TIFF. We might be able to do something like that with the OME Zarr writer once we get to that point.